hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryOperator.java

```diff
@@ -89,6 +89,13 @@ public final class EntryOperator {
     private boolean didMatchPredicate;
     private Data dataKey;
     private Object oldValue;
+    /**
+     * A clone of {@link #oldValue} that can be passed to an {@link EntryProcessor} to allow side-effect free mutations
```

Comment:
I think that's actually the other way around: `oldValue` is passed to the `EntryProcessor` and might be mutated by the EP#process implementation. 

`oldValueClone` is a pristine clone of the value originally stored in the record store before execution of the EP. It is used to populate the `oldValue` of an `EntryEvent`, if one needs to be fired. So `oldValueClone` is not actually passed to the `EntryProcessor`.